lib/clever_tap/entity.rb

```diff
@@ -0,0 +1,83 @@
+class CleverTap
+  class NoDataError < RuntimeError
+    def message
+      'No `data` param provided for Event'
+    end
+  end
+
+  class MissingIdentityError < RuntimeError
+    def message
+      "Couldn'n find `identity` in CleverTap.config or `data`"
+    end
+  end
+
+  class Entity
+    ALLOWED_IDENTITIES = %w(objectId FBID GPID).freeze
+    IDENTITY_STRING = 'identity'.freeze
+    TIMESTAMP_STRING = 'ts'.freeze
+    TYPE_KEY_STRING = 'type'.freeze
+    UPLOAD_LIMIT = 'Needs child class value'.freeze
```

Comment:
can you clarify this message?

Reply:
The normal functioning of the `to_h` of each child class depends on this constant being re-defined in the child class so I though leaving a message like that will help user locate the issue. Else he/she might be tricked into thinking that the limit should be defined in the parent class.